## Statistics exercises

### Exercise 1

In the Lady's Tea Test experiment, a subject is given eight cups of tea of sample size 8, of which 4 have been prepared with tea added first and four with milk added first. The cups are given in a random order and the subject is asked to determine which cup belongs to which group.

The odds of choosing the correct combination of cups at random are given as 1 in 70 possible combinations or 1.4%.

Task: Calculate the minimum number of cups of tea required to ensure the probability of randomly selecting the correct cups is less than or equal to 1%.

***

The odds of choosing the correct combination of cups increases with sample size *n* and the number of possible combinations, *k* being the number of cups to be chosen (half of *n*). The number of combinations for a sample size *n* can be calculated using the SciPy *comb* package:

In [17]:
from scipy.special import comb

k = 4
n = 8
combinations = comb(n, k, exact=True)
print("Possible combinations of k from n: " + str(combinations))

print("Chance of correct guess: " + str(round(100*(1/combinations),2)) + "%")

Possible combinations of k from n: 70
Chance of correct guess: 1.43%


Increasing *n* will lower the odds of guessing the correct combination at random to less than 1%:

In [18]:
k = 6
n = 12
combinations = comb(n,k, exact=True)
print("Possible combinations of k from n: " + str(combinations))

print("Chance of correct guess: " + str(round(100*(1/combinations),2)) + "%")

Possible combinations of k from n: 924
Chance of correct guess: 0.11%


Selecting 6 cups at random from a sample size of 12 yields 924 possible combinations and a 0.11% chance of guessing correctly.

Lowering *n* to 10 brings the probability closer to the 1% target:

In [19]:
k = 5
n = 10
combinations = comb(n,k, exact=True)
print("Possible combinations of k from n: " + str(combinations))

print("Chance of correct guess: " + str(round(100*(1/combinations),2)) + "%")

Possible combinations of k from n: 252
Chance of correct guess: 0.4%


Because the cups must remain discrete units and *n* and *k* must therefore remain integers, and assuming *k* must be exactly half of *n*, a sample size of 10 brings the probability closest to <=1%.

***

Bonus: How many would be required if you were to let the taster get one cup wrong while maintaining the 1% threshold?

***

Keeping a sample size of *n*=10, *k* is now set to 4 as the subject gets one guess wrong.

In [20]:
k = 4
n = 10
combinations = comb(n,k, exact=True)
print("Possible combinations of k from n: " + str(combinations))

print("Chance of correct guess: " + str(round(100*(1/combinations),2)) + "%")

Possible combinations of k from n: 210
Chance of correct guess: 0.48%


The probability of a correct guess is now at 0.48%, staying under the 1% target. Testing against the other values of *n*:

In [21]:
k = 3
n = 8
combinations = comb(n,k, exact=True)
print("Possible combinations of k from n: " + str(combinations))

print("Chance of correct guess: " + str(round(100*(1/combinations),2)) + "%")

Possible combinations of k from n: 56
Chance of correct guess: 1.79%


In [22]:
k = 5
n = 12
combinations = comb(n,k, exact=True)
print("Possible combinations of k from n: " + str(combinations))

print("Chance of correct guess: " + str(round(100*(1/combinations),2)) + "%")

Possible combinations of k from n: 792
Chance of correct guess: 0.13%


We can see that a sample size of 8 with 3 correct guesses is 1.79%, significantly over the 1% limit, while a sample size of 12 with 5 correct guesses has a probability of 0.13%.

***

### Exercise 2

Task: Use scipy's version of Fisher's exact test to simulate the Lady Tasting Tea problem.

***